go/stats/pull_backend.go

```diff
@@ -0,0 +1,27 @@
+package stats
+
+// PullBackend should be implemented to export pull metrics from Vitess.
```

Comment:
With the new change to just use the `stats.Publish` hook I actually don't think we need this interface any more since it looks like we never use it anywhere.

That also would let all the prom registration functions be private to the package instead of public.